test/test_catalog.py

```diff
@@ -0,0 +1,135 @@
+"""Unit Test for Catalog.py. Tests for download of schemas and catalog generation"""
+# file test_xmlmap/test_core.py
+#
+#   Copyright 2011 Emory University Libraries
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+#!/usr/bin/env python
+
+from __future__ import unicode_literals
+import os
+import unittest
+try:
+    from unittest import skipIf
+except ImportError:
+    from unittest2 import skipIf
+import glob
+try:
+    from urllib.request import urlretrieve
+    from urllib.request import urlopen
+except ImportError:
+    from urllib import urlretrieve
+    from urllib import urlopen
+import shutil
+from datetime import date
+from eulxml import xmlmap, __version__
+from lxml import etree
+from eulxml.catalog import Uri, Catalog
+
+
+
+class TestGenerateSchema(unittest.TestCase):
+    """:class:`TestGenerateSchema` class for Catalog testing"""
+    def setUp(self):
+        self.path = 'eulxml/test_schemas'
+        self.correct_schema = 'http://www.loc.gov/standards/mods/v3/mods-3-4.xsd'
+        self.wrong_schema = 'http://www.loc.gov/standards/mods/v3/mods34.xsd'
+        self.comment = 'Downloaded by eulxml %s on %s' % \
+              (__version__, date.today().isoformat())
+        # parseString wants a url. let's give it a proper one.
+        if not os.path.isdir(self.path):
+            os.mkdir(self.path)
+
+    def tearDown(self):
+        if os.path.isdir('eulxml/test_schemas'):
+            shutil.rmtree('eulxml/test_schemas')
+
+    def test_download_xml_schemas(self):
+        """Check if xsd schemas exist and download fresh copies """
+
+        #do files already exist
+        check_xsds = len(glob.glob(''.join([self.path, '*.xsd'])))
+        self.assertEqual(0, check_xsds)
+
+        #FIX THIS later
+        # try:
+        #     response_wrong = urlopen(self.wrong_schema)
+        # except:
+        #     print('Something went wrong')
+        #     response_error = urlopen(self.wrong_schema).getcode()
+        # expected, got = 404, response_error
+        # self.assertEqual(expected, got)
+
+        #downloading the right schemas
+        response_correct = urlopen(self.correct_schema)
```

Comment:
Shouldn't this be testing the download_schema method in eulxml.catalog ? It looks like instead you are duplicating the functionality.